File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> freq;
        vector<vector<int>> count(nums.size() + 1);
        for(int i = 0 ; i < nums.size() ; i++){
            freq[nums[i]]++;
        }  
        vector<int> res;
        for (const auto& pair : freq) {
            count[pair.second].push_back(pair.first) ;            
        }
        
         
        for(int i = count.size() - 1 ; i > 0 ; i--){            
            for(int n : count[i]){
                res.push_back(n);
                if(res.size() == k){
                    return res;
                }
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-8.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [1] * len(nums)
        prefix = 1

        for i in range(n):            
            res[i] = prefix
            prefix = prefix * nums[i]

        suffix = 1

        for i in range(n - 1, -1, -1):
            res[i] *= suffix
            suffix = suffix * nums[i]

        
        return res